coordinator/src/weaver.rs

```diff
@@ -81,7 +81,7 @@ impl Weaver {
 
         let linkables = self.modules.into_iter().map(|(_, link_info)| link_info.linkable.into_inner()).collect();
 
-        Ok((linkables, self.services.write().take().unwrap()))
+        Ok((linkables, self.services.write().take().expect("Services is an option type to return it here")))
```

Comment:
IMHO, use of `unwrap()` in production code (that is, outside of testing code) is a legitimate use of it. It means "I know what I'm doing. It won't be `Err` or `None`."

Besides the point, the sentence given as argument to `expect` is a bit weird to me. Shouldn't we say `services holds a value here"?

Reply:
1. I thought using `unwrap` is okay only when it is obvious. For example, it is checked in the code a few lines above.

When I read `unwrap` in code, I need to read how the value is created to ensure it will not panic.
I wanted to reduce the time to understand the `unwrap`.

I thought `weaver` is using `Option` in a non-obvious way. The `services` field is `Some` in the middle of `weave` function, and become `None` after the `weave` function.

How do you read code and what do you do when you read unknown `unwrap`?
I'm happy to discuss this with you. It will be great if we find a common consensus.

2. I agree with you in the point that the message of expect is bad. I think "services holds a value here" is not so good since it does not give information to programmers who read code. I'll think about the message when we agree on using `expect`.

Reply:
Currently weaver's use of `Option` is a bit unconventional. But that context should better be conveyed by some additional comments. To me, `expect()` is considered imposing the possibility of actual panic. So readers need to grok the message given to `expect()` to know no panic will be issued. But in the case of `unwrap()`, it tells readers no panic will occur.

Reply:
OK, I got your point. I'll let the code use `unwrap()`

Reply:
@dynaxis I fixed it: https://github.com/CodeChain-io/foundry/compare/b7c5340be5139bb75f41bb0e411f1935486bf0ab..df61a2256691d5d3a360b8919b0bcb8d33415334

Reply:
@majecty there is at least one more place correspnding to this discusson, and it's still using `expect()`.

Reply:
OK, I'll review the use cases.